clients/src/main/java/org/apache/kafka/common/serialization/ExceptionHandlingDeserializer.java

```diff
@@ -0,0 +1,107 @@
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.serialization;
+
+import org.apache.kafka.common.header.Headers;
+
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * When a {@link Deserializer} deserializes a message, it may throws an exception. E.g. a decryption deserializer fails to decrypt a corrupted message. When this happens, the {@link Deserializer} exits and stop consuming messages. This is often undesirable as it means the calling code is unable to take action, even reconnecting to the topic might just cause the exception to happen again.
+ * <p>
+ * {@link ExceptionHandlingDeserializer} is a decorator than allows you to wrap up another deserializer and return a result that contains either the deserialized result, or any exception throw by the deserializer.
+ * <p>
+ * You may configure using properties:
+ * <p>
+ * <code>
+ *     exception.handling.deserializer.delegate=org.apache.kafka.common.deserialization.ListDeserializer
+ * </code>
+ *
+ * @param <T>
+ */
+public class ExceptionHandlingDeserializer<T> implements Deserializer<ExceptionHandlingDeserializer.Result<T>> {
+
+    public static class Result<T> {
+        private final T result;
+        private final Exception exception;
+
+        Result(T result, Exception exception) {
+            this.result = result;
+            this.exception = exception;
+        }
+
+        public T getResult() {
+            return result;
+        }
+
+        public Exception getException() {
+            return exception;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Result<?> result1 = (Result<?>) o;
+            return Objects.equals(result, result1.result) && Objects.equals(exception, result1.exception);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(result, exception);
+        }
+    }
+
+    private Deserializer<T> delegate;
+
+    public ExceptionHandlingDeserializer() {
+    }
+
+    public ExceptionHandlingDeserializer(Deserializer<T> delegate) {
+        this.delegate = delegate;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public void configure(Map<String, ?> configs, boolean isKey) {
+        try {
+            delegate = (Deserializer<T>) Class.forName(String.valueOf(configs.get("exception.handling.deserializer.delegate"))).asSubclass(Deserializer.class).getDeclaredConstructor().newInstance();
```

Comment:
Let's pul `"exception.handling.deserializer.delegate"` config in `org.apache.kafka.clients.CommonClientConfigs` smth like `CommonClientConfigs.EXCEPTION_HANDLING_DESERIALIZER_DELEGATE`.

Reply:
Doing `String.valueOf(configs.get("exception.handling.deserializer.delegate")))` may cause `NullPointerException.

You might want to check if that config exists i.e.
```
Object delegate = configs.get("exception.handling.deserializer.delegate");
if (delegate == null) {
    ...
}
```